src/Renderer/Simulator/SimulatorTypes.fs

```diff
@@ -116,10 +116,33 @@ type SimulationData =
       // Keep track of the number of clock ticks of the simulation.
       ClockTickNumber: int }
 
+type SimulationErrorType =
+    | PortNumMissing
```

Comment:
This is a great idea. Some work to do it but safe and an improvement on existing code.

You should have a match function that prints all the error messages defined after the D.U. Where the error messages depend on data put that data in the D.U. Thus the error function is just SimulationErrorType -> string